pkg/daemon/ceph/osd/volume.go

```diff
@@ -1161,7 +1161,17 @@ func GetCephVolumeRawOSDs(context *clusterd.Context, clusterInfo *client.Cluster
 			if err != nil {
 				return nil, errors.Wrapf(err, "failed to get device info for %q", blockPath)
 			}
-			osd.DeviceClass = sys.GetDiskDeviceClass(diskInfo)
+			deviceType := sys.GetDiskDeviceType(diskInfo)
+			osd.DeviceType = deviceType
+			logger.Infof("setting device type %q for device %q", osd.DeviceType, diskInfo.Name)
+
+			crushDeviceClass := os.Getenv(oposd.CrushDeviceClassVarName)
+			if crushDeviceClass != "" {
+				osd.DeviceClass = crushDeviceClass
+			}
+			if osd.DeviceClass == "" {
+				osd.DeviceClass = deviceType
```

Comment:
why is deviceClass being set as deviceType over here?

Reply:
if user has not specify the device class, Rook will use the deviceType to apply the deviceClass